docs/weaviate/tutorials/tls-ssl.mdx

```diff
@@ -0,0 +1,955 @@
+
+- **Complexity level:** Novice
+- **Perfect for:** Small business, teams who prefer ClickOps (GUI-based operations), or personal projects.
+- **Infrastructure:** Docker and Docker Compose required. This approach is specifically for Docker-based deployments and is not compatible with Kubernetes.
+
+What you'll need:
+
+- A VPS or server with a public IP address
+- A registered domain name with DNS management access
+- Docker and Docker Compose installed
+- Basic command-line knowledge
+
+#### Step 1: Configure your DNS
+
+1. Before requesting SSL certificates, configure your DNS records:
+
+```
+A Record:    weaviate.yourdomain.com    → Your-Server-IP
+A Record:    grpc.yourdomain.com        → Your-Server-IP
+```
+
+:::tip
+Use a wildcard DNS record (`*.yourdomain.com`) for easier subdomain management.
+:::
+
+Make sure to wait for DNS propagation before proceeding.
+
+#### Step 2: Create your shared Docker network
+
+1. Create a network for secure container communication:
+
+```bash
+docker network create weaviate-network
+```
+
+:::info
+**Security Benefit:** By using a shared Docker network and not exposing Weaviate ports to the host, Weaviate is only accessible through the SSL-secured proxy, never directly from the internet.
+:::
+
+#### Step 3: Deploy Weaviate
+
+1. Create a directory for your project:
+
+```bash
+mkdir ~/weaviate-ssl
+cd ~/weaviate-ssl
+```
+
+2. Create a `docker-compose.yml` file for Weaviate.
+
+```yaml
+---
+services:
+  weaviate:
+    image: cr.weaviate.io/semitechnologies/weaviate:||site.weaviate_version||
+    container_name: weaviate
+    restart: unless-stopped
+    networks:
+      - weaviate-network
+    environment:
+      QUERY_DEFAULTS_LIMIT: 25
+      AUTHENTICATION_ANONYMOUS_ACCESS_ENABLED: "false"
+      PERSISTENCE_DATA_PATH: "/var/lib/weaviate"
+      DEFAULT_VECTORIZER_MODULE: "none"
+      CLUSTER_HOSTNAME: "node1"
+    volumes:
+      - weaviate_data:/var/lib/weaviate
+
+volumes:
+  weaviate_data:
+
+networks:
+  weaviate-network:
+    external: true
+```
+
+3. Start Weaviate:
+
+```bash
+docker-compose up -d
+```
+
+4. Verify that it's running:
+
+```bash
+# Test from within the Docker network (this should work)
+docker exec weaviate curl http://localhost:8080/v1/meta
+
+# Test from host (this should fail - connection refused)
+curl http://localhost:8080/v1/meta
+```
+
+:::tip
+**Expected Result:** The internal test succeeds, but the host test fails.
+
+This confirms Weaviate is only accessible within the Docker network, not from the host or internet.
+:::
+
+#### Step 4: Install Nginx Proxy Manager
+
+1. Create a separate directory:
+
+```bash
+mkdir ~/nginx-proxy-manager
+cd ~/nginx-proxy-manager
+```
+
+2. Create a `docker-compose.yml` file:
+
+```yaml
+services:
+  app:
+    image: "jc21/nginx-proxy-manager:latest"
+    container_name: nginx-proxy-manager
+    restart: unless-stopped
+    networks:
+      - weaviate-network
+    ports:
+      - "80:80" # HTTP
```

Comment:
Binding to port bellow 1024 requires root provilege on Linux - so I assume that this example would not work in Windows & Linux without being root. Probably worth binding to 8080:80.